dipy/viz/ui.py

```diff
@@ -1907,6 +1895,317 @@ def handle_events(self, actor):
                           self.handle_move_callback)
 
 
+class ListBox2D(UI):
+    """ UI component that allows the user to select items from a list.
+
+    Attributes
+    ----------
+    on_change: function
+        Callback function for when the selected items have changed.
+    """
+
+    def __init__(self, values, position, size,
+                 multiselection=True, reverse_scrolling=False,
+                 font_size=20, line_spacing=1.4):
+        """
+        Parameters
+        ----------
+        values: list of objects
+            Values used to populate this listbox. Objects must be castable
+            to string.
+        position: (float, float)
+            The initial position (x, y) in pixels.
+        size: (float, float)
+            The size of the system (x, y) in pixels.
+        multiselection: {True, False}
+            Whether multiple values can be selected at once.
+        reverse_scrolling: {True, False}
+            If True, scrolling up will move the list of files down.
+        font_size: int
+            The font size in pixels.
+        line_spacing: float
+            Distance between listbox's items in pixels.
+
+        """
+        super(ListBox2D, self).__init__()
+
+        self.size = size
+        self.values = values
+        self.font_size = font_size
+        self.line_spacing = line_spacing
+        self.multiselection = multiselection
+        self.reverse_scrolling = reverse_scrolling
+
+        self.slots = []
+        self.selected = set()
+
+        self.view_offset = 0
+        self._build()
+        self.set_center(position)
+        self.update()
+
+        # Define events captured by this UI component.
+        self.add_callback(self.up_button.actor, "LeftButtonPressEvent",
+                          self.up_button_callback)
+        self.add_callback(self.down_button.actor, "LeftButtonPressEvent",
+                          self.down_button_callback)
```

Comment:
In other UI elements I have separated the event handling logic in a separate function. This is just to keep the constructor a little clean.

Reply:
Good point.